hotshot-query-service/src/fetching/provider/query_service.rs

```diff
@@ -56,68 +55,56 @@ impl<Ver: StaticVersionType> QueryServiceProvider<Ver> {
     async fn fetch_legacy_payload<Types: NodeType>(
         &self,
+        payload_bytes: Vec<u8>,
+        common_bytes: Vec<u8>,
         req: PayloadRequest,
     ) -> Option<Payload<Types>> {
         let PayloadRequest(VidCommitment::V0(advz_commit)) = req else {
             return None;
         };
 
-        let res = try_join!(
-            self.client
-                .get::<ADVZPayloadQueryData<Types>>(&format!(
-                    "availability/payload/hash/{}",
-                    advz_commit
-                ))
-                .send(),
-            self.client
-                .get::<ADVZCommonQueryData<Types>>(&format!(
-                    "availability/vid/common/payload-hash/{}",
-                    advz_commit
-                ))
-                .send()
-        );
-        match res {
-            Ok((payload, common)) => {
-                let num_storage_nodes = ADVZScheme::get_num_storage_nodes(common.common()) as usize;
-                let bytes = payload.data.encode();
-                let commit = match advz_scheme(num_storage_nodes).commit_only(bytes) {
-                    Ok(commit) => commit,
-                    Err(err) => {
-                        tracing::error!(%err, "unable to compute VID commitment");
-                        return None;
-                    },
-                };
-                if commit != advz_commit {
-                    tracing::error!(?req, ?commit, "received inconsistent payload");
-                    return None;
-                }
+        let payload =
+            vbs::Serializer::<Ver>::deserialize::<ADVZPayloadQueryData<Types>>(&payload_bytes)
+                .map_err(|err| {
+                    tracing::error!(%err, "deserializing ADVZ payload. req={}", req.0);
+                })
+                .ok()?;
+
+        let common =
+            vbs::Serializer::<Ver>::deserialize::<ADVZCommonQueryData<Types>>(&common_bytes)
+                .map_err(|err| {
+                    tracing::error!(%err, "deserializing ADVZ common. req={}", req.0);
+                })
+                .ok()?;
+
+        let num_storage_nodes = ADVZScheme::get_num_storage_nodes(common.common()) as usize;
+        let bytes = payload.data.encode();
+
+        let commit = advz_scheme(num_storage_nodes)
+            .commit_only(bytes)
+            .map_err(|err| {
+                tracing::error!(%err, "unable to compute VID commitment");
+            })
+            .ok()?;
 
-                Some(payload.data)
-            },
-            Err(err) => {
-                tracing::error!("failed to fetch payload {req:?}: {err}");
-                None
-            },
+        if commit != advz_commit {
+            tracing::error!(?req, ?commit, "received inconsistent legacy payload");
+            return None;
         }
+
+        Some(payload.data)
     }
 
     async fn fetch_legacy_vid_common<Types: NodeType>(
         &self,
+        bytes: Vec<u8>,
```

Comment:
Similarly for this function

Reply:
https://github.com/EspressoSystems/espresso-network/pull/3175/commits/cc91b39dcb0248118394e8973fb26f5fa25429d0